scripts/create-secrets/k8sExecution.js

```diff
@@ -0,0 +1,320 @@
+            )
+            .then(function(ifCreate) {
+                if (!ifCreate) {
+                    console.log(
+                        chalk.yellow(
+                            `You need to create namespace \`${
+                                configData["cluster-namespace"]
+                            }\` before try again.`
+                        )
+                    );
+                    process.exit();
+                } else {
+                    createNamespace(configData["cluster-namespace"]);
+                }
+            });
+    }
+    return p.then(function() {
+        const namespace = configData["cluster-namespace"];
+        if (configData["use-cloudsql-instance-credentials"] === true) {
+            createFileContentSecret(
+                namespace,
+                "cloudsql-instance-credentials",
+                "credentials.json",
+                configData["cloudsql-instance-credentials"]
+            );
+        }
+        if (configData["use-storage-account-credentials"] === true) {
+            createFileContentSecret(
+                namespace,
+                "storage-account-credentials",
+                "db-service-account-private-key.json",
+                configData["storage-account-credentials"]
+            );
+        }
+        if (configData["use-smtp-secret"] === true) {
+            createSecret(namespace, "smtp-secret", {
+                username: configData["smtp-secret-username"],
+                password: configData["smtp-secret-password"]
+            });
+        }
+        (function() {
+            const data = {};
+            dbPasswordNames.forEach(key => {
+                data[key] = configData["db-passwords"];
+            });
+            createSecret(namespace, "db-passwords", data);
+        })();
+        if (configData["use-regcred"] === true) {
+            /**
+             * always use `regcred-password`
+             * `use-regcred-password-from-env` has been taken care seperately
+             */
+            createDockerRegistrySecret(
+                namespace,
+                "regcred",
+                configData["regcred-password"],
+                "registry.gitlab.com",
+                "gitlab-ci-token",
+                configData["regcred-email"]
+            );
+        }
+        if (
+            configData["use-oauth-secrets-google"] === true ||
+            configData["use-oauth-secrets-facebook"] === true
+        ) {
+            const data = {};
+            if (configData["use-oauth-secrets-google"]) {
+                data["google-client-secret"] =
+                    configData["oauth-secrets-google"];
+            }
+            if (configData["use-oauth-secrets-facebook"]) {
+                data["facebook-client-secret"] =
+                    configData["oauth-secrets-facebook"];
+            }
+            createSecret(namespace, "oauth-secrets", data);
+        }
+
+        (function() {
+            const data = {};
```

Comment:
Once again, why the function wrapper?

Reply:
As above, just want to create a new scope